File: backend/app/schemas/agent_tool.py
"""Agent tool schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class AgentToolInfo(BaseModel):
    id: UUID
    name: str
    label: str
    description: str | None = None
    tool_type: str
    config: dict
    enabled: bool
    sort: int
    created_at: datetime
    updated_at: datetime | None = None

    model_config = {"from_attributes": True}


class AgentToolCreate(BaseModel):
    name: str
    label: str
    description: str | None = None
    tool_type: str | None = "local"
    config: dict | None = None
    enabled: bool | None = True
    sort: int | None = 100


class AgentToolUpdate(BaseModel):
    name: str | None = None
    label: str | None = None
    description: str | None = None
    tool_type: str | None = None
    config: dict | None = None
    enabled: bool | None = None
    sort: int | None = None


class ToolCallLogInfo(BaseModel):
    id: UUID
    conversation_id: UUID | None = None
    tool_name: str
    arguments: dict
    result: str | None = None
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: ui/lib/api/agent-tool.ts
import { apiClient } from "./client";
import type { ApiSuccess } from "./auth";

export interface AgentToolInfo {
  id: string;
  name: string;
  label: string;
  description: string | null;
  tool_type: string;
  config: Record<string, unknown>;
  enabled: boolean;
  sort: number;
  created_at: string;
  updated_at: string | null;
}

export interface ToolCallLogInfo {
  id: string;
  conversation_id: string | null;
  tool_name: string;
  arguments: Record<string, unknown>;
  result: string | null;
  status: string;
  created_at: string;
}

export function listAgentTools() {
  return apiClient.get<ApiSuccess<AgentToolInfo[]>>("/agent-tools");
}

export function updateAgentTool(id: string, body: Partial<AgentToolInfo>) {
  return apiClient.put<ApiSuccess<AgentToolInfo>>(`/agent-tools/${id}`, body);
}

export function listToolCallLogs() {
  return apiClient.get<ApiSuccess<ToolCallLogInfo[]>>("/agent-tools/logs");
}
